enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/docverify/zenid/provider/ZenidDocumentVerificationProvider.java

```diff
@@ -91,30 +89,30 @@ public ZenidDocumentVerificationProvider(
     }
 
     @Override
-    public DocumentsSubmitResult checkDocumentUpload(OwnerId id, DocumentVerificationEntity document) throws DocumentVerificationException {
+    public DocumentsSubmitResult checkDocumentUpload(OwnerId id, DocumentVerificationEntity document) throws RemoteCommunicationException, DocumentVerificationException {
         DocumentsSubmitResult result = new DocumentsSubmitResult();
         ResponseEntity<ZenidWebUploadSampleResponse> responseEntity;
 
         try {
             responseEntity = zenidApiService.syncSample(document.getUploadId());
         } catch (RestClientException e) {
-            logger.warn("Failed REST call to check {} upload in ZenID, statusCode={}, responseBody='{}', {}",
-                    document, e.getStatusCode(), e.getResponse(), id, e);
-            throw new DocumentVerificationException("Unable to check document upload due to a REST call failure");
+            throw new RemoteCommunicationException(
```

Comment:
REST error leads to 503 REMOTE_COMMUNICATION_ERROR.
In addition, double logging is limited and the whole context is passed to the exception.